File: twitterEngine/src/spl/stomp/protocol/ErrorFrame.java
package spl.stomp.protocol;

/**
 * @name Error Frame
 * @author  Lidan Hifi
 * @author  Ken Saggy
 * 
 * ERROR stomp frame
 */
class ErrorFrame extends StompFrame {
    public ErrorFrame(String message) {
        super(StompCommand.ERROR);
        addFrameHeader("message", message);
    }
    
    public ErrorFrame(String message, String body) {
        super(StompCommand.ERROR, body);
        addFrameHeader("message", message);
    }
}

File: twitterClient/twitterClient/src/stomp/frames/ErrorFrame.cpp
//
//  ErrorFrame.cpp
//  Object to represent the ERROR frame type
//
//  Authors:    Ken Saggy
//              Lidan Hifi
//
//  Copyright 2013 Lidan Hifi, Ken Saggy

#include "../../../include/stomp/frames/ErrorFrame.h"

namespace stomp {
    ErrorFrame::ErrorFrame(): StompFrame(ERROR_FRAME_NAME) { }
    
    ErrorFrame::ErrorFrame(std::string message): StompFrame(ERROR_FRAME_NAME) {
        addHeaderPair("message", message);
    }
    
    ErrorFrame::ErrorFrame(std::string message, std::string body): StompFrame(ERROR_FRAME_NAME, body) {
        addHeaderPair("message", message);
    }
    
    void ErrorFrame::accept(FrameVisitor *visitor)
    {
        visitor->visit(this);
    }
}
